torch/_decomp/decompositions.py

```diff
@@ -4343,6 +4343,11 @@ def _weight_norm_interface(x, y, dim):
     norm = x.norm(2, keep_dim, keepdim=True)
     return x * (y / norm), norm
 
+@register_decomposition(aten.put)
+def put(self, index, source, accumulate=False):
```

Comment:
I was going to add this, but I noticed that `torch.put` doesn't support `out`

```
>>> torch.put(x, index, source, out=y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: put() got an unexpected keyword argument 'out'
```

Reply:
There is an out overload in the ATen level though, which is failing the tests currently because it's calling this decomposition with an out argument
```
RuntimeError: failed to run: aten.put.out
```